src/java_bytecode/java_types.h

```diff
@@ -18,6 +18,47 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/c_types.h>
 #include <util/optional.h>
 
+class java_annotationt : public irept
+{
+public:
+  class valuet : public irept
+  {
+  public:
+    valuet(irep_idt name, exprt value)
+    {
+      id(name);
+      set(ID_value, value);
+    }
+
+    const irep_idt &get_name()
+    {
+      return id();
+    }
+
+    const exprt &get_value()
+    {
+      return static_cast<const exprt &>(find(ID_value));
+    }
+  };
```

Comment:
I note that `java_annotationt` has no `id()` set?!

Reply:
That's in common with many ireps whose ID is implied by its position in the irep graph though, for example struct components.